controllers/storagecluster/external_resources.go

```diff
@@ -240,17 +240,28 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 		objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
 		switch d.Kind {
 		case "CephCluster":
-			monitoringIP, ok := d.Data["MonitoringEndpoint"]
-			if !ok || monitoringIP == "" {
+			monitoringIP := d.Data["MonitoringEndpoint"]
+			if monitoringIP == "" {
 				err := fmt.Errorf(
 					"Monitoring Endpoint not present in the external cluster secret %s",
 					externalClusterDetailsSecret)
 				r.Log.Error(err, "Failed to get Monitoring IP.")
 				return err
 			}
+			// replace any comma in the monitoring ip string with space
+			// and then collect individual (non-empty) items' array
+			monIPArr := strings.Fields(strings.ReplaceAll(monitoringIP, ",", " "))
 			monitoringPort := d.Data["MonitoringPort"]
 			if monitoringPort != "" {
```

Comment:
This may be outside this PR (or at least a separate commit in this PR), but is there a reason we don't do `checkEndpointReachable()` if `monitoringPort == ""`? Is there no default port to fall back on, or should we consider it invalid to not provide a port?

Reply:
If port is not given, we (from OCS-Operator) are not providing a default. Rook will internally provide a default port when the given port is empty.

Reply:
:+1: